src/screens/DistrictMatch/DistrictMatch.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+import { View, StyleSheet, ActivityIndicator } from 'react-native';
+// Views
+import {ZipCode, Address, Success, Failure} from './screens';
+// API Method & Controller
+import fetchDistrict from './fetchDistrict';
+
+import Mixins from '../../styles/mixins';
+import colors from '../../styles/colors';
+
+
+class DistrictMatch extends Component {
+  constructor(props){
+    super(props);
+    this.fetchDistrict = fetchDistrict.bind(this);
+  }
+
+  state = {
+    zipcode: '',
+    address: '',
+    city: '',
+    isLoading: false,
+    view:'ENTER_ZIPCODE',
+    district: '',
+    error: ''
+  };
+
+  submit = () => {
+    this.setState({isLoading: true, error: ''}, () => {
+      this.fetchDistrict();
+    })
+  }
+
+  handleChange = (input) => {
+    this.setState({...input});
+  }
+
+  renderScreen = () => {
+    const methods = {
+      handleChange: this.handleChange,
+      submit: this.submit,
+      navigate: this.props.navigate
+    };
+    switch(this.state.view){
```

Comment:
Nice refactor - moving all these components to their own files makes this way cleaner